config/confighttp/confighttp.go

```diff
@@ -478,16 +480,18 @@ func (hss *ServerConfig) ToServer(_ context.Context, host component.Host, settin
 		next:            handler,
 		includeMetadata: hss.IncludeMetadata,
 	}
+	errorLog, err := zap.NewStdLogAt(settings.Logger, zapcore.ErrorLevel)
```

Comment:
We should return errors as soon as they happen, not later.

```suggestion
	errorLog, err := zap.NewStdLogAt(settings.Logger, zapcore.ErrorLevel)
	if err != nil {
		 return nil, err
	}
```

Reply:
the suggested change was the  initial code too
 but in 
 https://github.com/open-telemetry/opentelemetry-collector/pull/11830/files/71c16e0b293b4d6ed25146c4dd295093e672afd7#r1880527497
 it was suggested to return server,err so that tests can pass

Reply:
Then, how about something like:

```golang
server := &http.Server{
	Handler:           handler,
	ReadTimeout:       hss.ReadTimeout,
	ReadHeaderTimeout: hss.ReadHeaderTimeout,
	WriteTimeout:      hss.WriteTimeout,
	IdleTimeout:       hss.IdleTimeout,
}

errorLog, err := zap.NewStdLogAt(settings.Logger, zapcore.ErrorLevel)
if err != nil {
	return server, err
}

server.ErrorLog = errorLog
return server, nil
```

Then, we can handle multiple things that could return errors, and we also keep error return alongside its definition.

Reply:
I will check the report locally and may be make the change according to it 
the command for it is `make gotest-with-cover` right?